scheduler/pkg/kafka/gateway/worker.go

```diff
@@ -301,17 +302,28 @@ func (iw *InferWorker) restRequest(ctx context.Context, job *InferWork, maybeCon
 	return iw.produce(ctx, job, iw.topicNamer.GetModelTopicOutputs(job.modelName), b, false, extractHeadersHttp(response.Header))
 }
 
+// Add all external headers to request metadata
+func (iw *InferWorker) addMetadataToOutgoingContext(ctx context.Context, job *InferWork) context.Context {
```

Comment:
Could we add a smaller test to check which headers are supposed to be returned? Serves also as a kind of documentation.

Reply:
added